pkg/apis/projectcalico/v3/ippool.go

```diff
@@ -78,8 +78,19 @@ type IPPoolSpec struct {
 	// Deprecated: this field is only used for APIv1 backwards compatibility.
 	// Setting this field is not allowed, this field is for internal use only.
 	NATOutgoingV1 bool `json:"nat-outgoing,omitempty" validate:"omitempty,mustBeFalse"`
+
+	// AllowedUse controls what the IP pool will be used for.  If not specified or empty, defaults to
```

Comment:
"if not specified or empty" - do we want to retain "empty" in case we have some reason to let users create IP pools with no allowed uses?

e.g., if the pools are created solely for NAT and not for IP address allocation?

Reply:
On second thought, I think we already have `disabled` for this, and a distinction between `empty` and `nil` is always confusing in these sorts of APIs, so let's not have that distinction.